src/widgets/unfold_widgets/preprocessing_widget.py

```diff
@@ -200,3 +222,42 @@ def remove_nulls_warning(self):
 
     def handle_warning_click(self, button):
         self.data_submitted = "OK" in button.text()
+
+    def reduce_dimensions(self, dim_number=None):
+        self.engine.state.imported_data.drop(self.engine.state.reduced_columns, axis=1, inplace=True, errors='ignore')
+        self.engine.state.raw_data.drop(self.engine.state.reduced_columns, axis=1, inplace=True)
+        self.engine.state.reduced_columns = self.engine.reduce_dimensions(dim_number)
+        self.mark_reduced_columns = True
+        self.show_reduction_results()
+
+    def set_reduction_bounds(self, max_dimensions):
+        self.num_dimensions_spinbox.setMinimum(2)
```

Comment:
Why we deny reduce to 1 dimension?

Reply:
Honestly, was set to 2 arbitrary because I thought that linear shape then wouldn't be interesting but I agree that it can make sense because we keep information about many parameters in one dimension, so I'll change that

Reply:
I agree that 2 dimension is better, bacause we can visualize it. But users should can reduce to 1 dim.